diff --git a/dist/lib/generate-output.js b/dist/lib/generate-output.js
index 12fac90dbb22435d4294dfec8943725bb79132de..fe68e8dbbf18019664e4dae5cdaca34fa5492967 100644
--- a/dist/lib/generate-output.js
+++ b/dist/lib/generate-output.js
@@ -28,6 +28,9 @@ async function generateOutput(html, relativePath, config, browserRef) {
     }
     const browser = await getBrowser();
     const page = await browser.newPage();
+    if (config.launch_options?.timeout) {
+        page.setDefaultNavigationTimeout(config.launch_options.timeout);
+    }
     const urlPathname = (0, path_1.join)(relativePath, 'index.html').split(path_1.sep).join(path_1.posix.sep);
     // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
     await page.goto(`http://localhost:${config.port}/${urlPathname}`); // make sure relative paths work as expected
diff --git a/dist/lib/get-html.js b/dist/lib/get-html.js
index 5fe1eb6319e21d16f4b3ed65d458cf12c68c3d7b..b386237cf90b43ec6bab8c79f3eba0c031ae0993 100644
--- a/dist/lib/get-html.js
+++ b/dist/lib/get-html.js
@@ -7,10 +7,26 @@ const get_marked_with_highlighter_1 = require("./get-marked-with-highlighter");
  */
 const getHtml = (md, config) => `<!DOCTYPE html>
 <html>
-	<head><title>${config.document_title}</title><meta charset="utf-8"></head>
+	<head>
+		<title>${config.document_title}</title>
+		<meta charset="utf-8">
+		<script>
+			// 给 highlightjs-line-numbers.js 添加挂载点，防止报错
+			window.hljs = {};
+		</script>
+		<script src="https://unpkg.com/highlightjs-line-numbers.js@2.8.0/dist/highlightjs-line-numbers.min.js"></script>
+	</head>
 	<body class="${config.body_class.join(' ')}">
 		${(0, get_marked_with_highlighter_1.getMarked)(config.marked_options, config.marked_extensions)(md)}
 	</body>
+	<script>
+		Array.prototype.forEach.call(document.querySelectorAll('pre code'), e => {
+			e.classList.add('hljs');
+		});
+		window.hljs.initLineNumbersOnLoad({
+			singleLine: true
+		});
+	</script>
 </html>
 `;
 exports.getHtml = getHtml;
diff --git a/dist/lib/get-marked-with-highlighter.js b/dist/lib/get-marked-with-highlighter.js
index 0aa7089cada3394eca3cab8a31a49e5eb12b7cbd..71f6e81bec0f91f1365929a660ec14281da320a4 100644
--- a/dist/lib/get-marked-with-highlighter.js
+++ b/dist/lib/get-marked-with-highlighter.js
@@ -8,7 +8,12 @@ const highlight_js_1 = __importDefault(require("highlight.js"));
 const marked_1 = require("marked");
 const getMarked = (options, extensions) => {
     marked_1.marked.setOptions(Object.assign({ highlight: (code, languageName) => {
+            // 没有设置语言，或则是一下语言的，需要自动检测语言
+            const autoLang = ['txt', 'text', 'plain', 'plaintext'];
             const language = highlight_js_1.default.getLanguage(languageName) ? languageName : 'plaintext';
+            if (autoLang.includes(language)) {
+                return highlight_js_1.default.highlightAuto(code).value;
+            }
             return highlight_js_1.default.highlight(code, { language }).value;
         }, langPrefix: 'hljs ' }, options));
     marked_1.marked.use(...extensions);